### ISSUE TITLE
Fcall Optimizations: Issues

### ISSUE THREAD

**Issue by steffengy** [phalcon/zephir#867]
As already commented on 33dd09662e1dd85d4c6b85d433ca1e5bc58906c7
I can observe the following problem:

- memory "leak" (memory usage grows with additional method calls)

(Not anymore important) Fixed stuff:
- a strange win32 bug occuring on **ZEPHIR_FCALL_TYPE_ZVAL_METHOD**
  resulting in a crash in zend_mm_set_custom_handlers (broken zval or something which fails gc?)
Do you have any knowledge of these problems
The crash currently really is not easy to debug:
- bypassing the new fcall-method for ZEPHIR_FCALL_TYPE_ZVAL_METHOD resolves the crash
- it does not happen in win32-debug builds nor on linux
- valgrind on linux also does not report anything (in release and debug builds)
- it's a pretty huge code base (>10 k LOC) and I haven't managed to find a specific
  reason which might cause the crash.
TLDR: Optimized implementation probably not quite perfect yet ^^

**Comment by steffengy** [comment]
Crash seems to be resolved by 0d420b55c3c4f00a82967b99c9536488e195e973
But memory "leak"/"loss" still occuring.

**Comment by steffengy** [comment]
Since the crash is resolved, creating a followup ticket with steps to reproduce for the memory leak.
